File: src/main/scripts/Messages.js
db.createCollection("messages", {
    validator: {
        $jsonSchema: {
            bsonType: "object",
            required: [ "sender", "receiver", "subject", "content", "sentDate" ],
            properties: {
                sender: {
                    bsonType: "string"
                },
                receiver: {
                    bsonType: "string"
                },
                subject: {
                    bsonType: "string"
                },
                content: {
                    bsonType: "string"
                },
                sentDate: {
                    bsonType: "date"
                },
                status: {
                    enum: [ "READ", "UNREAD" ]
                }
            }
        }
    }
})

File: src/main/java/com/aldren/messaging/document/Messages.java
package com.aldren.messaging.document;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Getter
@Setter
@Document(collection = "messages")
public class Messages {

    @Id
    private String id;

    private String sender;

    private String receiver;

    private String subject;

    private String content;

    private Date sentDate;

    private String status;

}
